src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -50,7 +53,28 @@ function updateHostComponent(current, workInProgress) {
   console.log('host component', workInProgress.type, typeof workInProgress.pendingProps.children === 'string' ? workInProgress.pendingProps.children : '');
 
   var nextChildren = workInProgress.pendingProps.children;
-  reconcileChildren(current, workInProgress, nextChildren);
+
+  let priority = workInProgress.pendingWorkPriority;
+  if (workInProgress.pendingProps.hidden && priority !== OffscreenPriority) {
```

Comment:
Yea. This is temporarily hacked in place, it will defer to the hostConfig to figure this out in the future. E.g. the DOM can look for `props.style.display === 'none'`.